src/test/java/spoon/test/template/TemplateTest.java

```diff
@@ -442,6 +439,25 @@ public void testCheckBoundTemplate() throws Exception {
 		assertEquals(injectedCode, method.getBody().getStatement(0));
 	}
 
+	@Test
+	public void testAbstractTemplateIsWellFormed() {
+		// contract: IsWellFormed method of AbstractTemplate class returns false if there are zero template parameter
+
+		// arrange
+		AbstractTemplate abstractTemplate = new AbstractTemplate() {
+			@Override
+			public CtElement apply(CtType targetType) {
```

Comment:
Avoid using [raw types](https://docs.oracle.com/javase/tutorial/java/generics/rawTypes.html).

```suggestion
		AbstractTemplate<CtElement> abstractTemplate = new AbstractTemplate<CtElement>() {
			@Override
			public CtElement apply(CtType<?> targetType) {
```